src/htm/encoders/SimHashDocumentEncoder.cpp

```diff
@@ -127,33 +217,24 @@ namespace htm {
     output.setDense(result.getDense());
   } // end method encode
 
-  /**
-   * Encode (Alternate calling style: Simple list method)
-   * @see SimHashDocumentEncoder.hpp
-   * @see encode(const std::map<std::string, UInt> input, SDR &output)
-   * @see encode(std::string input, SDR &output)
-   */
-  void SimHashDocumentEncoder::encode(const std::vector<std::string> input, SDR &output)
-  {
-    std::map<std::string, UInt> inputWeighted;
-    for (const auto& token : input) {
-      inputWeighted[token] = 1u;
-    }
-    encode(inputWeighted, output);
-  } // end method encode (list alternate)
-
   /**
    * Encode (Alternate calling style: Simple string method)
    * @see SimHashDocumentEncoder.hpp
-   * @see encode(const std::map<std::string, UInt> input, SDR &output)
    * @see encode(const std::vector<std::string> input, SDR &output)
    */
   void SimHashDocumentEncoder::encode(const std::string input, SDR &output)
   {
+    NTA_CHECK(input.length() > 0u)
+      << "Encoding input string should have at least lenth 1.";
```

Comment:
maybe don't crash and just return `output.zero()` on an empty input? I think some other encoder does that.

Reply:
done: https://github.com/htm-community/htm.core/blob/3e0eee3415503ec532db7363dd02c5a4e77b76ce/src/htm/encoders/SimHashDocumentEncoder.cpp#L115-L118